### ISSUE TITLE
Synthetic data has 25%-50% repeated values

### ISSUE THREAD

**Issue by nejadhav** [sdv-dev/CTGAN#250]
### Environment details

If you are already running CTGAN, please indicate the following details about the environment in
which you are running it:

* CTGAN version: 0.5.1
* Python version: 3.9.12
* Operating System:Windows 10 

### Problem description

Every time I try to generate synthetic data, I get 25% to 50% repeated values. I had 120 data points & I have generated 1200 data points using CTGAN. How can I avoid getting repeated values in the new data?

I am sharing data description of original & synthetic data for understanding the data summary

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/u1255401/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/u1255401/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link=blue vlink=purple>

ORIGINAL DATA
--
  | YEAR | clm1 | clm2
count | 120 | 1.20E+02 | 1.20E+02
mean | 2017.992 | 2.31E+08 | 4.41E+04
std | 1.894245 | 6.03E+08 | 1.58E+05
min | 2015 | 8.33E+04 | 8.30E+01
25% | 2016 | 7.82E+06 | 3.95E+03
50% | 2018 | 3.97E+07 | 1.02E+04
75% | 2020 | 1.62E+08 | 2.48E+04
max | 2021 | 4.79E+09 | 1.57E+06

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/u1255401/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/u1255401/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link=blue vlink=purple>

SYNTHETIC DATA
--
  | YEAR | clm1 | clm2
count | 1200 | 1.20E+03 | 1200
mean | 2019.273 | 1.90E+08 | 31692.76
std | 1.696263 | 5.18E+08 | 62662.81
min | 2015 | 8.33E+04 | 83.04
25% | 2018 | 8.33E+04 | 83.04
50% | 2020 | 7.82E+06 | 18941.13
75% | 2021 | 1.57E+08 | 42160.16
max | 2021 | 4.69E+09 | 644243.8

</body>

</html>

</body>

</html>

### What I already tried

I tried changing seed, epochs, batch size , reduced the number of rows generated, changed the data, but nothing really worked. I am getting repeated values in at least 1 column.

here's my code:
torch.manual_seed(40)
np.random.seed(40)
model = CTGAN(epochs=400)
model.fit(data)
new_data = model.sample(num_rows=1000,randomize_samples=0,batch_size=300)

**Comment by npatki** [comment]
Hi @nejadhav, nice to meet you! Thanks for sharing the details. Just to make sure I'm understanding correctly, I see that in the synthetic data stats:
- For `YEAR`: The 75th percentile and MAX are the same, meaning that 25% of the data has the same value
- For `clm1` and `clm2`: The MIN and 25th percentile are the same, meaning 25% of the data has the same value

Is this correct? 

The CTGAN does work better with more epochs so I would suggest trying a larger number such as 1,000 to see if the results are improving. The exact answer may depend on the nature of your data.

I'm curious what the histogram of the real data looks like for each of these columns. For example: are there outliers? How many unique values are in the real data to begin with? And how was this data collected? 

Generally speaking, GANs will work better when you have many training examples and the data represents natural observations from the real world. If there is a particular use of synthetic data where you'd want to control or enforce certain properties from a smaller dataset, I'd also suggest exploring the [GaussianCopula Model](https://sdv.dev/SDV/user_guides/single_table/gaussian_copula.html) from the SDV library.

**Comment by npatki** [comment]
Hello, seems like it's been a while since we've touched this issue. I'll close it as completed, since we have some suggestions for how to move forward.

If you're still running into this issue, please feel free to reply and I can reopen the issue for investigation.
